onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.cc

```diff
@@ -543,24 +543,21 @@ Status ModelBuilder::Compile(std::unique_ptr<Model>& model) {
 
     bool all_ops_supported = std::all_of(supported_ops, supported_ops + num_nnapi_ops_,
                                          [](bool is_supported) { return is_supported; });
-    // TODO, To allow fallback to CPU if it's not strict CPU_DISABLED mode
     if (!all_ops_supported) {
       // There are some ops not supported by the list of the target devices
       // Fail the Compile
       //
       // TODO, add some logic to not fail for some cases
       // Such as, if there are some acceptable fall back to CPU (nnapi-reference)
       // and CPU is not in the target devices list
-      return ORT_MAKE_STATUS(ONNXRUNTIME, EP_FAIL,
-                             "The model cannot run using the current set of target devices, ",
-                             GetDevicesDescription(nnapi_target_devices_));
-    }
-
-    // Workaround bugs in NNAPI drives on some phones
-    // where ops are passed checking by 'ANeuralNetworksModel_getSupportedOperationsForDevices'
-    // but failed at compilation.
-    // ANeuralNetworksCompilation_create allows falling back to CPU if compilation fails
-    if (target_device_option_ != TargetDeviceOption::ALL_DEVICES) {
+        return ORT_MAKE_STATUS(ONNXRUNTIME, EP_FAIL,
+                               "The model cannot run using the current set of target devices, ",
+                               GetDevicesDescription(nnapi_target_devices_));
+    } else if (target_device_option_ != TargetDeviceOption::ALL_DEVICES) {
```

Comment:
nit: don't need else after return
```suggestion
    }
    
    if (target_device_option_ != TargetDeviceOption::ALL_DEVICES) {
```